Kata 3.13.0 dependency bump

stevenhorsman opened the pull request:
Bump kata-containers, guest-components and trustee versions in response to them being bumped in the kata-containers 3.13.0 release.

wainersm reviewed (approved):
And let the release show begin!

stevenhorsman commented:
So unless I've done something wrong it looks like the updates aren't working for us as the tests that require the KBS are all failing...

mkulke commented:
> So unless I've done something wrong it looks like the updates aren't working for us as the tests that require the KBS are all failing

yeah, the `HTTP/1.1 500 Internal Server Error` doesn't look good, logs for guest-components should hopefully reveal more.

mkulke commented:
> This is a bit confusing as locally I've tried too different machines incase I messed something up and the keyrelease test passed both times:
> 
> ```
> === NAME  TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test
>     assessment_runner.go:549: Deleting pod kbs-key-release...
>     assessment_runner.go:556: Pod kbs-key-release has been successfully deleted within 60s
> --- PASS: TestLibvirtKbsKeyRelease (195.43s)
>     --- PASS: TestLibvirtKbsKeyRelease/DoTestKbsKeyReleaseForFailure_test (104.63s)
>         --- PASS: TestLibvirtKbsKeyRelease/DoTestKbsKeyReleaseForFailure_test/Kbs_key_release_is_failed (9.53s)
>     --- PASS: TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test (90.50s)
>         --- PASS: TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test/Kbs_key_release_is_successful (5.42s)
> ```

if you used the CI podvm image in the local tests the only likely culprit could be KBS, no?

maybe we should add kbs logs to the debug output in the CI

stevenhorsman commented:
> > This is a bit confusing as locally I've tried too different machines incase I messed something up and the keyrelease test passed both times:
> > ```
> > === NAME  TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test
> >     assessment_runner.go:549: Deleting pod kbs-key-release...
> >     assessment_runner.go:556: Pod kbs-key-release has been successfully deleted within 60s
> > --- PASS: TestLibvirtKbsKeyRelease (195.43s)
> >     --- PASS: TestLibvirtKbsKeyRelease/DoTestKbsKeyReleaseForFailure_test (104.63s)
> >         --- PASS: TestLibvirtKbsKeyRelease/DoTestKbsKeyReleaseForFailure_test/Kbs_key_release_is_failed (9.53s)
> >     --- PASS: TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test (90.50s)
> >         --- PASS: TestLibvirtKbsKeyRelease/KbsKeyReleasePod_test/Kbs_key_release_is_successful (5.42s)
> > ```
> 
> if you used the CI podvm image in the local tests the only likely culprit could be KBS, no?
> 
> maybe we should add kbs logs to the debug output in the CI

I've worked out what went wrong in my 2nd test (which I guess was in the first too). My kustomization update for the CAA didn't get picked up, probably as I did it too late. When I did a kubectl edit on it I can reproduce. I wasn't expecting the CAA to be the issue though, more the podvm though?

stevenhorsman commented:
Okay, I'm even more confused now. I re-ran the key release test to check the kbs log and it passed even with my double checking that it's using the PR's CAA, so maybe I was wrong last time. I've added debug for the kbs into the workflow, but it won't show up here because of the pull_request_target though, so I might end up splitting that into a separate PR.

stevenhorsman commented:
Ok locally I managed to reproduce on a re-run and the kbs logs:
```
[2025-01-17T16:42:03Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/resource/caa/workload_key/test_key.bin HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.005185
[2025-01-17T16:42:03Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/resource-policy HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.001718
[2025-01-17T16:42:03Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/attestation-policy HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.003227
[2025-01-17T16:43:34Z INFO  kbs::http::attest] Auth API called.
[2025-01-17T16:43:34Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/auth HTTP/1.1" 200 74 "-" "attestation-agent-kbs-client/0.1.0" 0.002916
[2025-01-17T16:43:34Z INFO  kbs::http::attest] Attest API called.
[2025-01-17T16:43:34Z INFO  attestation_service] Sample Verifier/endorsement check passed.
[2025-01-17T16:43:34Z ERROR kbs::http::error] Attestation failed: Policy Engine evaluation failed: Policy evaluation denied for default
[2025-01-17T16:43:34Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/attest HTTP/1.1" 401 183 "-" "attestation-agent-kbs-client/0.1.0" 0.005624
[2025-01-17
```

fitzthum commented:
Ok looks like that's coming from the AS policy rather than the KBS policy. Let's make sure we're using the new default policy.

fitzthum commented:
Hm yeah also it doesn't look like that `PolicyDenied` error is actually used anymore. That run may have been using an older version.

stevenhorsman commented:
Ok - here are the new kbs logs that look more useful:
```
[2025-01-17T17:37:38Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/resource/caa/workload_key/test_key.bin HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.007317
[2025-01-17T17:37:38Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/resource-policy HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.005282
[2025-01-17T17:37:38Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/attestation-policy HTTP/1.1" 200 0 "-" "kbs-client/0.1.0" 0.001877
[2025-01-17T17:39:08Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/auth HTTP/1.1" 200 74 "-" "attestation-agent-kbs-client/0.1.0" 0.003937
[2025-01-17T17:39:08Z INFO  attestation_service] Sample Verifier/endorsement check passed.
[2025-01-17T17:39:08Z ERROR kbs::error] AttestationError(RcarAttestFailed { source: verify TEE evidence failed

    Caused by:
        At least one policy claim must be set. })
[2025-01-17T17:39:08Z INFO  actix_web::middleware::logger] 10.244.1.1 "POST /kbs/v0/attest HTTP/1.1" 401 140 "-" "attestation-agent-kbs-client/0.1.0" 0.013499
```

fitzthum commented:
Ok, looks like that is still an issue with the AS policy. Maybe we're using the one for the simple token or something. Let me see.

fitzthum commented:
Yeah, [this function](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/test/provisioner/trustee_kbs.go#L473-L486) isn't really correct. We don't want to use the `allow_all` policy with the attestation service. Fortunately, I think for now we can just never call it and only use the default policy and things will be fine.

stevenhorsman commented:
I've removed the set-attestation-policy allow-all calls to pick up the default. It looks like we were using this to check that some requests were rejected setting deny_all, so I've switched those to use the resource-policy instead?

stevenhorsman commented:
Progress - just sealed secret tests failing now, so I'll check what I've missed there

stevenhorsman commented:
It doesn't look like there are any obvious KBS errors, but the secret isn't being unsealed and shows: 
```
assessment_helpers.go:233: Pod log: sealed.fakejwsheader.eyJhbm5vdGF0aW9ucyI6e30sIm5hbWUiOiJrYnM6Ly8vY2FhL3dvcmtsb2FkX2tleS90ZXN0X2tleS5iaW4iLCJwcm92aWRlciI6ImticyIsInByb3ZpZGVyX3NldHRpbmdzIjp7fSwidHlwZSI6InZhdWx0IiwidmVyc2lvbiI6IjAuMS4wIn0=.fakesignature
```
so I guess it's the same padding issue we saw in kata-containers. Now to work out how to remove that...

fitzthum reviewed (approved):
LGTM
